### PR TITLE
Refactor controller, library

### PR DESCRIPTION
**Pull request by sin3point14**
Mostly a refactor, no functional/logical changes other than UI now showing the full url/path of the song to be played, but that will be fixed in a future PR

tested manually for all 3 platforms

### CODE REVIEW

File: shared/src/commonMain/kotlin/MediaPlayerController.kt [hunk 1]
```diff
@@ -1,10 +1,6 @@
 // Thanks to https://github.com/SEAbdulbasit/MusicApp-KMP/
```
Comment (wryonik) [inline shared/src/commonMain/kotlin/MediaPlayerController.kt]: ye har file mein kyu daal rakha hai, rm

File: composeApp/src/commonMain/kotlin/App.kt [hunk 2]
```diff
@@ -35,11 +35,20 @@ object GlobalState {
...
 fun App(
     mediaPlayerController: MediaPlayerController,
-    rootPicker: suspend () -> String?
+    makeLibrary: suspend () -> Library?
 ) {
     val navController: NavHostController = rememberNavController()
     var songsList: List<String> = emptyList()
     var currentSong: String? = null
+    var library: Library? by remember { mutableStateOf(null) }
+    LaunchedEffect(Unit) {
```
Comment (wryonik) [inline composeApp/src/commonMain/kotlin/App.kt:44]: Note for later: we need to use something else on which this `LaunchedEffects` is dependent for better performance

File: shared/src/androidMain/kotlin/AndroidController.kt [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+    override fun seek(): Long? {
+        println(player.currentPosition)
+        println(player.duration)
+        println(player.contentDuration)
+        println(player.contentPosition)
+        println(player.currentMediaItem)
+        println("player.currentMediaItem")
+
+
```
Comment (wryonik) [inline shared/src/androidMain/kotlin/AndroidController.kt:68]: ye mera hai but rm 🥲

File: shared/src/wasmJsMain/kotlin/WebController.kt [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+class WebController: MediaPlayerController {
+    private val audioElement = document.createElement("audio") as HTMLAudioElement
+    override fun prepare(
+        songUri: String,
+        listener: MediaPlayerListener
+    ) {
+        println("listening to")
+        println(songUri)
+        audioElement.src = songUri
+        audioElement.addEventListener("canplaythrough", {
```
Comment (wryonik) [inline shared/src/wasmJsMain/kotlin/WebController.kt:13]: we should add enums for these random keys. We need to add enums for routes as well

File: shared/src/wasmJsMain/kotlin/WebController.kt [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    override fun mediaDuration(): Long? {
+        return audioElement.duration.toLong()
+    }
+
+    override fun setTime(time: Long) {
+//        mediaPlayer?.controls()?.setTime(time)
+        audioElement.currentTime = time.toDouble()
+    }
+
+    override fun release() {
```
Comment (wryonik) [inline shared/src/wasmJsMain/kotlin/WebController.kt:56]: Note for Shubham: Check what this function does and why is it empty?

File: shared/src/wasmJsMain/kotlin/WebLibrary.kt [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+    override fun nextTrack() {
+        if (songsList.isNotEmpty() && currentSongIndex != null) {
+            currentSongIndex = (currentSongIndex!! + 1) % songsList.size
+        }
+    }
+
+    override fun previousTrack() {
+        if (songsList.isNotEmpty() && currentSongIndex != null) {
+            currentSongIndex = (currentSongIndex!! - 1 + songsList.size) % songsList.size
```
Comment (wryonik) [inline shared/src/wasmJsMain/kotlin/WebLibrary.kt:30]: Ye negative mein mod nahi leta kya sahi se!!

File: shared/src/commonMain/kotlin/MediaPlayerController.kt (hunk 1 above)
Comment (sin3point14) [reply shared/src/commonMain/kotlin/MediaPlayerController.kt]: removed in a65a251e7e289c80e6c73fd7a31f94e1cc9e21c1

File: shared/src/androidMain/kotlin/AndroidController.kt (hunk 3 above)
Comment (sin3point14) [reply shared/src/androidMain/kotlin/AndroidController.kt:68]: will do in future pr, Opened issue: https://github.com/HH-0rg/Museli/issues/3

File: composeApp/src/commonMain/kotlin/App.kt (hunk 2 above)
Comment (sin3point14) [reply composeApp/src/commonMain/kotlin/App.kt:44]: Maybe open an issue

File: shared/src/wasmJsMain/kotlin/WebController.kt (hunk 4 above)
Comment (sin3point14) [reply shared/src/wasmJsMain/kotlin/WebController.kt:13]: sure, but lets keep that out of this PR, maybe create an issue

File: shared/src/wasmJsMain/kotlin/WebController.kt (hunk 5 above)
Comment (sin3point14) [reply shared/src/wasmJsMain/kotlin/WebController.kt:56]: create issue

File: shared/src/wasmJsMain/kotlin/WebLibrary.kt (hunk 6 above)
Comment (sin3point14) [reply shared/src/wasmJsMain/kotlin/WebLibrary.kt:30]: real shit https://stackoverflow.com/questions/47455006/kotlin-negative-modulo-returns-negative-value

**Review by wryonik — APPROVED** [review]

**Pull request merged**